### ISSUE TITLE
Add a lint for style rules whose selectors are just "&"

### ISSUE THREAD

**Issue by nex3** [stylelint-scss/stylelint-scss#394]
Sometimes mixin authors who don't fully understand how parent selectors work will write `&` as the sole selector for a style rule rather than including properties in the root level of the mixin. We should add a lint to warn them away from this practice. For example:

```scss
@mixin add-color($color) {
  & {
    color: $color;
  }
}
```

is semantically identical to:

```scss
@mixin add-color($color) {
  color: $color;
}
```

/cc @rambleraptor

**Comment by kristerkari** [comment]
I'm thinking that could it be part of the existing rule for selectors (even if it's related to mixins)? 

https://github.com/kristerkari/stylelint-scss/blob/master/src/rules/selector-no-redundant-nesting-selector/README.md

**Comment by nex3** [comment]
Yeah, that looks like it would make sense.

**Comment by pamelalozano16** [comment]
Can we close this out since this example works with `selector-no-redundant-nesting-selector`? Here's a [Playground example](https://stylelint.io/demo/#N4Igxg9gJgpiBcIACBXAzjABAHRGghmmvBgDYxgAuEATrgNzYB2zSAtgJYAeHTm+UKAFpIpWgAoAJKNoBKTMGbZKAOgBmECJgVLKylQCN8NTADIdLPVZk14mAMRqADE8aXlAX11emP1px4+AWEAdwALDkoYKRt5RXdKGztHNTU3PyZ2bl5+QSEDUnwwAGsYiDEaON1zeOVlJIcXV29mDxAAGhA1DnIAOXw2OEQYLgGAB3IVNDAiDvAIJm6AcwQQeO1cEaimKDRcOwBtZm0NvEoAT3JSXkoRBeWhNEp8HeNhaaJcY8wAXXbv3A0FDkPYICwnHB4GZoAD0ZAo1BoQiYECENBgUBQryYtyYMCevCWjxg5CotH2mEoQJg3x8bU6kEWHCWADFaGx8JRVgArNALOawMZoVbrSFPS4km4U3CFKJPXD-Phii5XG53JlEp4vKBvaUgWX4vQdAFnCXXHHqh5a140d7Q-YyzmGr6+EAeIA).

**Comment by nex3** [comment]
@pamelalozano16 that playground example doesn't show any problems for me.
